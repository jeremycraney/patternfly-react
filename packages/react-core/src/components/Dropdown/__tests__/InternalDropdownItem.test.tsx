import * as React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { InternalDropdownItem } from '../InternalDropdownItem';
import { DropdownSeparator } from '../DropdownSeparator';

describe('dropdown items', () => {
  test('a', () => {
    const view = render(<InternalDropdownItem>Something</InternalDropdownItem>);
    expect(view.container).toMatchSnapshot();
  });

  test('button', () => {
    const view = render(<InternalDropdownItem component="button">Something</InternalDropdownItem>);
    expect(view.container).toMatchSnapshot();
  });

  test('separator', () => {
    const view = shallow(<DropdownSeparator />);
    expect(view).toMatchSnapshot();
  });

  describe('hover', () => {
    test('a', () => {
      const view = render(<InternalDropdownItem isHovered>Something</InternalDropdownItem>);
      expect(view.container).toMatchSnapshot();
    });
    test('button', () => {
      const view = render(
        <InternalDropdownItem isHovered component="button">
          Something
        </InternalDropdownItem>
      );
      expect(view.container).toMatchSnapshot();
    });
  });

  describe('disabled', () => {
    test('a', () => {
      const view = render(<InternalDropdownItem isDisabled>Something</InternalDropdownItem>);
      expect(view.container).toMatchSnapshot();
    });
    test('button', () => {
      const view = render(
        <InternalDropdownItem isDisabled component="button">
          Something
        </InternalDropdownItem>
      );
      expect(view.container).toMatchSnapshot();
    });
  });

  describe('aria-disabled', () => {
    test('a', () => {
      const view = render(<InternalDropdownItem isAriaDisabled>Something</InternalDropdownItem>);
      expect(view.container).toMatchSnapshot();
    });
    test('button', () => {
      const view = render(
        <InternalDropdownItem isAriaDisabled component="button">
          Something
        </InternalDropdownItem>
      );
      expect(view.container).toMatchSnapshot();
    });
  });

  describe('description', () => {
    test('a', () => {
      const view = render(
        <InternalDropdownItem description="Something's link description">Something</InternalDropdownItem>
      );
      expect(view.container).toMatchSnapshot();
    });
    test('button', () => {
      const view = render(
        <InternalDropdownItem description="Something's button description" component="button">
          Something
        </InternalDropdownItem>
      );
      expect(view.container).toMatchSnapshot();
    });
  });
});
