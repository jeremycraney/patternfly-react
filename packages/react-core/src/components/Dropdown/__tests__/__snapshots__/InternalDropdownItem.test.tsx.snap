// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown items a 1`] = `
<div>
  <li
    role="none"
  >
    <a
      class=""
    >
      Something
    </a>
  </li>
</div>
`;

exports[`dropdown items aria-disabled a 1`] = `
<div>
  <li
    role="none"
  >
    <a
      aria-disabled="true"
      class="pf-m-aria-disabled"
    >
      Something
    </a>
  </li>
</div>
`;

exports[`dropdown items aria-disabled button 1`] = `
<div>
  <li
    role="none"
  >
    <button
      aria-disabled="true"
      class="pf-m-aria-disabled"
      type="button"
    >
      Something
    </button>
  </li>
</div>
`;

exports[`dropdown items button 1`] = `
<div>
  <li
    role="none"
  >
    <button
      class=""
      type="button"
    >
      Something
    </button>
  </li>
</div>
`;

exports[`dropdown items description a 1`] = `
<div>
  <li
    role="none"
  >
    <a
      class="pf-m-description"
    >
      <div
        class="pf-c-dropdown__menu-item-main"
      >
        Something
      </div>
      <div
        class="pf-c-dropdown__menu-item-description"
      >
        Something's link description
      </div>
    </a>
  </li>
</div>
`;

exports[`dropdown items description button 1`] = `
<div>
  <li
    role="none"
  >
    <button
      class="pf-m-description"
      type="button"
    >
      <div
        class="pf-c-dropdown__menu-item-main"
      >
        Something
      </div>
      <div
        class="pf-c-dropdown__menu-item-description"
      >
        Something's button description
      </div>
    </button>
  </li>
</div>
`;

exports[`dropdown items disabled a 1`] = `
<div>
  <li
    role="none"
  >
    <a
      aria-disabled="true"
      class=""
    >
      Something
    </a>
  </li>
</div>
`;

exports[`dropdown items disabled button 1`] = `
<div>
  <li
    role="none"
  >
    <button
      aria-disabled="true"
      class=""
      type="button"
    >
      Something
    </button>
  </li>
</div>
`;

exports[`dropdown items hover a 1`] = `
<div>
  <li
    role="none"
  >
    <a
      class=""
    >
      Something
    </a>
  </li>
</div>
`;

exports[`dropdown items hover button 1`] = `
<div>
  <li
    role="none"
  >
    <button
      class=""
      type="button"
    >
      Something
    </button>
  </li>
</div>
`;

exports[`dropdown items separator 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
